src/components/ConnectWallet/ConnectWalletDialog.tsx

```diff
@@ -0,0 +1,229 @@
+  const [qrCodeSvg, setQrCodeSvg] = useState<string>('')
+
+  useEffect(() => {
+    let stale = false
+    if (qrUri) {
+      toQrCodeSvg(qrUri).then((qrCodeSvg) => {
+        if (stale) return
+        setQrCodeSvg(qrCodeSvg)
+      })
+    } else {
+      walletConnect.activate(defaultChainId).catch((e) => {
+        if (stale) return
+        onError?.(e)
+      })
+    }
+    return () => {
+      stale = true
+    }
+  }, [qrUri, walletConnect, defaultChainId, onError])
+
+  useEffect(() => {
+    const disconnectListener = async (err: Error | null, _: any) => {
+      if (err) onError?.(err)
+      // Clear saved QR URI after disconnection
+      setQrUri(undefined)
+      walletConnect.deactivate()
+    }
+    walletConnect.provider?.connector.on('disconnect', disconnectListener)
+
+    // Need both URI event listeners
+    walletConnect.events.on(URI_AVAILABLE, async (uri: string) => {
+      if (uri) {
+        setQrUri(uri)
+      }
+    })
+
+    const uriListener = async (err: Error | null, payload: any) => {
+      if (err) onError?.(err)
+      const uri: string = payload.params[0]
+      if (uri) {
+        setQrUri(uri)
+      }
+    }
+    walletConnect.provider?.connector.on('display_uri', uriListener)
+
+    return () => {
+      walletConnect.events.off(URI_AVAILABLE)
+      ;(walletConnect.provider?.connector as unknown as EventEmitter | undefined)?.off('display_uri', uriListener)
+    }
+  })
+
+  return (
+    <StyledMainButton onClick={onClick}>
+      <StyledMainButtonRow>
+        <ButtonContents>
+          <img src={logoSrc} alt={walletName} width={32} />
+          <ThemedText.Subhead1>
+            <Trans>{walletName}</Trans>
+          </ThemedText.Subhead1>
+          <ThemedText.Caption color="secondary">
+            <Trans>Scan to connect your wallet. Works with most wallets.</Trans>
+          </ThemedText.Caption>
+        </ButtonContents>
+        <QRCodeWrapper dangerouslySetInnerHTML={{ __html: qrCodeSvg }} />
+      </StyledMainButtonRow>
+    </StyledMainButton>
+  )
+}
+
+function MetaMaskButton({ walletName, logoSrc, onClick }: ButtonProps) {
+  return (
+    <StyledSmallButton onClick={onClick}>
+      <ButtonContents>
+        <img src={logoSrc} alt={walletName} width={26} />
+        <ThemedText.Subhead1>
+          <Trans>{walletName}</Trans>
+        </ThemedText.Subhead1>
+      </ButtonContents>
+    </StyledSmallButton>
+  )
+}
+
+function NoWalletButton() {
+  const helpCenterUrl = 'https://help.uniswap.org/en/articles/5391585-how-to-get-a-wallet'
+  return (
+    <StyledSmallButton onClick={() => window.open(helpCenterUrl)}>
+      <StyledNoWalletText>
+        <Trans>I don&apos;t have a wallet</Trans>
+      </StyledNoWalletText>
+    </StyledSmallButton>
+  )
+}
+
+export function ConnectWalletDialog() {
+  let defaultConnections: Web3Connection[]
+  const [firstConnector] = connections[0]
+  if (firstConnector instanceof EIP1193 || firstConnector instanceof Url) {
```

Comment:
name a variable for what's inside the parens, then you won't need a comment below